test/cluster/regression/github-15496.td

```diff
@@ -0,0 +1,29 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Regression test for https://github.com/MaterializeInc/materialize/issues/15496
+# This test is special in that it requires MZ_SOFT_ASSERTIONS=0 at present.
+
+# Set data for test up
+> SET cluster = cluster1;
+
+> CREATE TABLE base (data bigint, diff bigint);
+
+> CREATE MATERIALIZED VIEW data AS SELECT data FROM base, repeat_row(diff);
+
+> INSERT INTO base VALUES (1, 1);
+
+> INSERT INTO base VALUES (1, -1), (1, -1);
+
+# Run a query that would generate a panic before the fix. Note that
+# we expect the query to succeed for now, but follow-up work might
+# eventually lead us to favor a SQL-level error for such a query, as
+# tracked by https://github.com/MaterializeInc/materialize/issues/17178
+> SELECT SUM(data), MAX(data) FROM data;
+<null> <null>
```

Comment:
Is the best-effort handling of aggregate now going to introduce `null` values in places where before those were not possible (for example `SELECT SUM(y) FROM t GROUP BY x`)?

Reply:
Outside of the reduce collation, the behavior was already to avoid outputting when we had negative accumulations. Here is the handling in hierarchical aggregates: https://github.com/MaterializeInc/materialize/blob/0cf899f8358c7daa19d5fa146b1c6c220f4a18e2/src/compute/src/render/reduce.rs#L613 as well as https://github.com/MaterializeInc/materialize/blob/0cf899f8358c7daa19d5fa146b1c6c220f4a18e2/src/compute/src/render/reduce.rs#L660

And here is the handing in basic aggregates: https://github.com/MaterializeInc/materialize/blob/0cf899f8358c7daa19d5fa146b1c6c220f4a18e2/src/compute/src/render/reduce.rs#L530

So AFAICT the change is transforming a crash into eliminating output, and only in the case of reduce collations.